DataFormats/SiPixelDetId/interface/PixelChannelIdentifier.h

```diff
@@ -4,29 +4,17 @@
 #include <utility>
 #include "FWCore/Utilities/interface/GCC11Compatibility.h"
 
-class PixelChannelIdentifier {
-public:
-  typedef unsigned int PackedDigiType;
-  typedef unsigned int ChannelType;
-
-  static std::pair<int, int> channelToPixel(int ch) {
-    int row = (ch >> thePacking.column_width) & thePacking.row_mask;
-    int col = ch & thePacking.column_mask;
-    return std::pair<int, int>(row, col);
-  }
-
-  static int pixelToChannel(int row, int col) { return (row << thePacking.column_width) | col; }
-
+namespace pixelchanelidentifierimpl {
```

Comment:
The `Packing` class had to be removed from within `PixelChannelIdentifier` for a technicality about C++ and `constexpr`.  You can not declare a `constexpr` member data where the type is defined within the class. The reason is the class is not _defined_ until its closing `};` and that includes classes defined within it. The last bit is you can't use `constexpr` on an undefined type. Declaring `Packing` outside of `PixelChannelIdentifier` makes use `Packing` is fully defined when the compiler reaches the `constexpr`.